@here/olp-sdk-core/lib/utils/MultiPartUploadWrapper.ts

```diff
@@ -177,9 +221,211 @@ export class MultiPartUploadWrapper {
+        let offset = 0;
+        let chunkNumber = 0;
+        let uploadedChunks = 0;
+
+        while (unreadBytes > 0) {
+            const promises = [];
+
+            while (promises.length < parallelRequests) {
+                const buffer = await blobData.readBytes(offset, chunkSize);
+
+                if (buffer.byteLength === 0) {
+                    break;
+                }
+
+                chunkNumber++;
+                offset += buffer.byteLength;
+                unreadBytes -= buffer.byteLength;
+
+                promises.push(
+                    request
+                        .uploadPart({
+                            data: buffer,
+                            layerId: opts.layerId,
+                            multipartToken:
+                                startMultipartResponse.multipartToken,
+                            billingTag: opts.billingTag,
+                            contentLength: buffer.byteLength,
+                            contentType: opts.contentType,
+                            partNumber: chunkNumber,
+                            url: startMultipartResponse.uploadPartUrl
+                        })
+                        .then(uploadPartResponse => {
+                            parts.push({
+                                id: uploadPartResponse.id,
+                                number: chunkNumber
+                            });
+
+                            uploadedChunks++;
+
+                            opts.onStatus &&
+                                opts.onStatus({
+                                    chunkId: uploadPartResponse.id,
+                                    chunkNumber,
+                                    chunkSize: buffer.byteLength,
+                                    totalChunks,
+                                    uploadedChunks
+                                });
+                        })
+                );
+            }
+
+            await Promise.all(promises);
+        }
+
+        await request.completeMultipartUpload({
+            parts,
+            url: startMultipartResponse.completeUrl,
+            multipartToken: startMultipartResponse.multipartToken,
+            layerId: opts.layerId,
+            billingTag: opts.billingTag
+        });
+
+        blobData.finally && blobData.finally();
+
         return STATUS_CODES.NO_CONTENT;
     }
+
+    private isStartMultipartResponseValid(obj: {
+        multipartToken?: string;
+        uploadPartUrl?: string;
+        completeUrl?: string;
+        statusUrl?: string;
+    }): boolean {
+        return (
+            obj.multipartToken !== undefined ||
+            (obj.uploadPartUrl !== undefined && obj.completeUrl !== undefined)
+        );
+    }
+
+    private isBlobData(input: any): boolean {
+        return (
+            typeof input.size === "function" &&
+            typeof input.readBytes === "function"
+        );
+    }
+
+    private async prepareBlobData(
```

Comment:
```suggestion
    private async getBlobData(
```
as I don't see much preparation in there except creating an instance of BlobData from different types.